src/main/kotlin/Network.kt

```diff
@@ -24,5 +24,50 @@ class Network(val capacity: Int, val listOfCars: MutableList<Car>) {
             car.isDelayed = overCapacity
         }
     }
-
+    fun CarsDevidedByCapacity():Double{
+        val x = (listOfCars.size / capacity).toDouble()
+        return x
+    }
+fun switchCase(x:Double):Double {
+    var y :Double = 0.0
+    if (x > 1.5) {
+        y = 0.9
+    }
+    if ((1.25 <= x) && (x <= 1.5)) {
+        y = 0.7
+    }
+    if ((1.1 <= x) && (x <= 1.25)) {
+        y = 0.5
+    }
+    if ((0.9 <= x) && (x <= 1.1)) {
+        y = 0.3
+    }
+    if ((0.7 <= x) && (x <= 0.9)) {
+        y = 0.2
+    }
+    if ((0.5 <= x) && (x <= 0.7)) {
+        y = 0.1
+    }
+    if (x < 0.5) {
+        y = 0.05
+    }
+    return y
+}
+    fun applyingDelay(ListOfCarsInterestToDriveInteger:MutableList<Int>,y:Double):MutableList<Int>{
+        var ListOfCarsAfterDelay:MutableList<Int> = mutableListOf()
+        for (car in ListOfCarsInterestToDriveInteger){
+            if (car==1) {
+                val z = Math.random()
+                var value = 1
+                if (y > z) {
+                    value = 0
+                }
+                ListOfCarsAfterDelay.add(ListOfCarsInterestToDriveInteger.get(car) * value)
```

Comment:
Here you could set value as false and instead of multiplying you could use && to keep the variables in Boolean and don't have to convert to Integer and back.